tests/console/validate_fstab.pm

```diff
@@ -0,0 +1,54 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate that the mounting option of each partition
+# defined in test data, is set in fstab is as expected.
+# Example of test_data syntax:
+#
+# test_data:
+#   disks:
+#     - name: vda
+#       partitions:
+#         - name: vda1
+#           mounting_options:
+#             mount_point: swap
+#           fstab_options:
+#             mount_by: UUID
+#         - name: vda2
+#           mounting_options:
+#             mount_point: /
+#           fstab_options:
+#             mount_by: "Device Name"
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use scheduler 'get_test_suite_data';
+use partitions_validator_utils 'validate_mounting_option';
+
+sub run {
+    my $test_data = get_test_suite_data;
+    select_console 'root-console';
+
+    foreach my $disk (@{$test_data->{disks}}) {
+        foreach my $partition (@{$disk->{partitions}}) {
+            if ($partition->{fstab_options}) {
+                validate_mounting_option({
+                        partition   => $partition->{name},
+                        mount_by    => $partition->{fstab_options}->{mount_by},
+                        mount_point => $partition->{mounting_options}{mount_point}});
```

Comment:
One `->` is missing.

Reply:
I thought we are supposed to avoid using stuff when they are not necessary :boom:

Reply:
strange, is not a hash ref? I was thinking that your validation was outdated, wondering why you don't need to arrow here...

Reply:
me too :rofl:

Reply:
ooh ooh! I just learnt something in Perl, Thanks Sofia, so only the first arrow is needed by Perl to understand that is a reference (I have live a while without knowing that, interesting), for our test data all our structure are ref, so we can in fact omit them all the time, only in the case that we mix ref structure with no ref structure is when forcing to put the arrow would make sense.
Might I ask you to have it in uniform way, so you omit them in the whole file?
Good lesson here, and you're right, not carpal tunnel syndrome need it, we can omit them I believe.